src/shared/utilities/collectionUtils.ts

```diff
@@ -160,3 +160,52 @@ export async function take<T>(sequence: AsyncIterable<T>, count: number): Promis
 
     return result
 }
+
+// TODO: Move this to a function?
+// TODO: Allow this to generate its own client?
+export class IteratingAWSCall<TRequest, TResponse> {
```

Comment:
Yeah, that's more or less what I was getting at, this is v1 and I wasn't 100% sure how the state management worked within the iterator (and I knew that this did indeed work). Next steps are error handling and test writing, assuming that some of the implementation details are agreed upon (especially since this is representing a departure from how we're doing most of our client calls; no iterators, passing in raw requests returning raw responses, etc.).

Reply:
guess I am confused about `AsyncGenerator<TResponse>` being the return type.